jenkins/Jenkinsfile

```diff
@@ -0,0 +1,268 @@
+//"Update the custom data proc image used in leo.  The name of the job contains the name of the dataproc image created. This job will fail if master of terra-docker has not been updated since the last run, as the git hash has not changed. This is a happy side-effect of the fact image names must be unique."
+
+//populated by first step
+def ArrayList<String> images = []
+def shortHash = ""
+
+def shouldAbortBuild = false
+def LinkedHashMap<String, String> imageToParamMap = [:]
+def branchName = "automated-custom-image-hash-update-"
+
+def getShortHash(hash) {
+  return hash.substring(0,7)
+}
+
+def boolean hasHashChanges(ArrayList<String> images) {
+  def Set imagesChanged = []
+
+  def String[] configPaths = readFile("jenkins/confLocations.txt").trim().split("\n")
+
+
+  for (String configFile: configPaths) {
+
+    println("in change detection loop, before bash. File: $configFile") 
+    def gitOutput = sh(
+      //TODO: revert
```

Comment:
Is this TODO accurate?

Reply:
the ~60 in the line below is needed for testing or it wont detect any changes. It should be ~1 in the final version.